Tests/SymbolKitTests/UnifiedGraph/GraphCollectorTests.swift

```diff
@@ -0,0 +1,107 @@
+                            ],
+                            relationships: [])
+        
+        let b = SymbolGraph(metadata: .init(formatVersion: .init(major: 1, minor: 0, patch: 0), generator: "unit-test"),
+                            module: .init(name: "B", platform: .init()),
+                            symbols: [
+                                .init(identifier: .init(precise: "s:BB", interfaceLanguage: "swift"),
+                                      names: .init(title: "B", navigator: nil, subHeading: nil, prose: nil),
+                                      pathComponents: ["B"],
+                                      docComment: nil,
+                                      accessLevel: .init(rawValue: "public"),
+                                      kind: .init(parsedIdentifier: .class, displayName: "Class"),
+                                      mixins: [:])
+                            ],
+                            relationships: [])
+        
+        let a_At_B = SymbolGraph(metadata: .init(formatVersion: .init(major: 1, minor: 0, patch: 0), generator: "unit-test"),
+                            module: .init(name: "A", platform: .init()),
+                            symbols: [
+                                .init(identifier: .init(precise: "s:BBAatB", interfaceLanguage: "swift"),
+                                      names: .init(title: "AatB", navigator: nil, subHeading: nil, prose: nil),
+                                      pathComponents: ["B", "AatB"],
+                                      docComment: nil,
+                                      accessLevel: .init(rawValue: "public"),
+                                      kind: .init(parsedIdentifier: .class, displayName: "Class"),
+                                      mixins: [:])
+                            ],
+                            relationships: [])
+        
+        
+        // test with default strategy (extension graphs get attached to extend**ed** graph
+        var collector = GraphCollector()
+
+        collector.mergeSymbolGraph(a, at: .init(fileURLWithPath: "A.symbols.json"))
+        collector.mergeSymbolGraph(b, at: .init(fileURLWithPath: "B.symbols.json"))
+        collector.mergeSymbolGraph(a_At_B, at: .init(fileURLWithPath: "A@B.symbols.json"))
+        
+        var (unifiedGraphs, _) = collector.finishLoading()
+        
+        XCTAssertEqual(unifiedGraphs.count, 2)
+        
+        for graph in unifiedGraphs.values {
+            if graph.moduleName == "A" {
+                XCTAssertEqual(graph.symbols.count, 1)
+                XCTAssertTrue(graph.symbols.keys.contains("s:AA"))
+            } else if graph.moduleName == "B" {
+                XCTAssertEqual(graph.symbols.count, 2)
+                XCTAssertTrue(graph.symbols.keys.contains("s:BB"))
+                XCTAssertTrue(graph.symbols.keys.contains("s:BBAatB"))
+            } else {
+                XCTFail()
+            }
+        }
```

Comment:
I feel like, rather than looping over the graphs like this, it would be more easily understood if you pulled each graph out separately:

```suggestion
        XCTAssertEqual(unifiedGraphs.count, 2)

        let graphA = try XCTUnwrap(unifiedGraphs["A"])
        XCTAssertEqual(graphA.symbols.count, 1)
        XCTAssert(graphA.symbols.keys.contains("s:AA"))

        let graphB = try XCTUnwrap(unifiedGraphs["B"])
        XCTAssertEqual(graphB.symbols.count, 2)
        XCTAssert(graphB.symbols.keys.contains("s:BB"))
        XCTAssert(graphB.symbols.keys.contains("s:BBAtB"))
```

The same goes for the other loop when using the other merge strategy.

Reply:
👍